wp-includes/sqlite-ast/class-wp-sqlite-driver.php

```diff
@@ -0,0 +1,2592 @@
+		$success = false;
+		try {
+			if ( 0 === $this->transaction_level ) {
+				$this->execute_sqlite_query( 'BEGIN' );
+			} else {
+				$this->execute_sqlite_query( 'SAVEPOINT LEVEL' . $this->transaction_level );
+			}
+			$success = $this->last_exec_returned;
+		} finally {
+			if ( $success ) {
+				++$this->transaction_level;
+				if ( function_exists( 'do_action' ) ) {
+					/**
+					 * Notifies that a transaction-related query has been translated and executed.
+					 *
+					 * @param string $command The SQL statement (one of "START TRANSACTION", "COMMIT", "ROLLBACK").
+					 * @param bool $success Whether the SQL statement was successful or not.
+					 * @param int $nesting_level The nesting level of the transaction.
+					 *
+					 * @since 0.1.0
+					 */
+					do_action( 'sqlite_transaction_query_executed', 'START TRANSACTION', (bool) $this->last_exec_returned, $this->transaction_level - 1 );
+				}
+			}
+		}
+		return $success;
+	}
+
+	/**
+	 * Commit the current transaction or nested transaction.
+	 *
+	 * @return boolean True on success, false on failure.
+	 */
+	public function commit() {
+		if ( 0 === $this->transaction_level ) {
+			return false;
+		}
+
+		--$this->transaction_level;
+		if ( 0 === $this->transaction_level ) {
+			$this->execute_sqlite_query( 'COMMIT' );
+		} else {
+			$this->execute_sqlite_query( 'RELEASE SAVEPOINT LEVEL' . $this->transaction_level );
+		}
+
+		if ( function_exists( 'do_action' ) ) {
+			do_action( 'sqlite_transaction_query_executed', 'COMMIT', (bool) $this->last_exec_returned, $this->transaction_level );
+		}
+		return $this->last_exec_returned;
+	}
+
+	/**
+	 * Rollback the current transaction or nested transaction.
+	 *
+	 * @return boolean True on success, false on failure.
+	 */
+	public function rollback() {
+		if ( 0 === $this->transaction_level ) {
+			return false;
+		}
+
+		--$this->transaction_level;
+		if ( 0 === $this->transaction_level ) {
+			$this->execute_sqlite_query( 'ROLLBACK' );
+		} else {
+			$this->execute_sqlite_query( 'ROLLBACK TO SAVEPOINT LEVEL' . $this->transaction_level );
+		}
+		if ( function_exists( 'do_action' ) ) {
+			do_action( 'sqlite_transaction_query_executed', 'ROLLBACK', (bool) $this->last_exec_returned, $this->transaction_level );
+		}
+		return $this->last_exec_returned;
+	}
+
+	/**
+	 * Executes a MySQL query in SQLite.
+	 *
+	 * @param string $query The query.
+	 *
+	 * @throws Exception If the query is not supported.
+	 */
+	private function execute_mysql_query( WP_Parser_Node $ast ) {
+		if ( 'query' !== $ast->rule_name ) {
+			throw new Exception( sprintf( 'Expected "query" node, got: "%s"', $ast->rule_name ) );
+		}
+
+		$children = $ast->get_child_nodes();
+		if ( count( $children ) !== 1 ) {
+			throw new Exception( sprintf( 'Expected 1 child, got: %d', count( $children ) ) );
+		}
+
+		$ast = $children[0]->get_child_node();
+		switch ( $ast->rule_name ) {
+			case 'selectStatement':
+				$this->query_type = 'SELECT';
```

Comment:
let's make sure we can handle union selects

Reply:
I added tests for both UNION and UNION ALL.